mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksViewModel.kt

```diff
@@ -415,68 +410,29 @@ abstract class MavericksViewModel<S : MavericksState>(
         lifecycleOwner: LifecycleOwner? = null,
         deliveryMode: DeliveryMode,
         action: suspend (T) -> Unit
-    ): Job {
-        val flow = if (lifecycleOwner == null || MavericksTestOverrides.FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER) {
-            this
-        } else if (deliveryMode is UniqueOnly) {
-            val lastDeliveredValue: T? = lastDeliveredValue(deliveryMode)
-            this
-                .assertOneActiveSubscription(lifecycleOwner, deliveryMode)
-                .dropWhile { it == lastDeliveredValue }
-                .flowWhenStarted(lifecycleOwner)
-                .distinctUntilChanged()
-                .onEach { lastDeliveredStates[deliveryMode.subscriptionId] = it }
-        } else {
-            flowWhenStarted(lifecycleOwner)
-        }
-
-        val scope = (lifecycleOwner?.lifecycleScope ?: viewModelScope) + configFactory.subscriptionCoroutineContextOverride
-        return scope.launch(start = CoroutineStart.UNDISPATCHED) {
+    ): Job = if (lifecycleOwner != null) {
+        collectLatest(
+            lifecycleOwner,
+            lastDeliveredStates,
+            activeSubscriptions,
+            deliveryMode,
+            action
+        )
+    } else {
+        (viewModelScope + configFactory.subscriptionCoroutineContextOverride).launch(start = CoroutineStart.UNDISPATCHED) {
             // Use yield to ensure flow collect coroutine is dispatched rather than invoked immediately.
             // This is necessary when Dispatchers.Main.immediate is used in scope.
             // Coroutine is launched with start = CoroutineStart.UNDISPATCHED to perform dispatch only once.
             yield()
-            flow.collectLatest(action)
-        }
-    }
-
-    @Suppress("EXPERIMENTAL_API_USAGE")
-    private fun <T> Flow<T>.assertOneActiveSubscription(owner: LifecycleOwner, deliveryMode: UniqueOnly): Flow<T> {
```

Comment:
I think I'm missing something, where did this code get moved to?

Reply:
omg, sorry. i had moved it to a new file and it wasnt tracked/committed